backend/internal/flow/flowservice.go

```diff
@@ -212,12 +207,26 @@ func (s *FlowService) loadContextFromStore(flowID, actionID string, inputData ma
 	return &ctx, nil
 }
 
+// prepareContext prepares the flow context by merging any data.
+func prepareContext(ctx *model.EngineContext, inputData map[string]string) {
+	// Append any input data present to the context
+	if len(inputData) > 0 {
+		ctx.UserInputData = sysutils.MergeStringMaps(ctx.UserInputData, inputData)
+	}
+}
+
+// removeContext removes the flow context from the store.
+func (s *FlowService) removeContext(flowID string, logger *log.Logger) {
+	s.mu.Lock()
+	delete(s.store, flowID)
+	s.mu.Unlock()
```

Comment:
```suggestion
	s.mu.Unlock()
```
Shall we move this to a defer block?

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/0f228aba36a6f1e1a8f24ad70d100e743c0c0d69..ec961583c3fa551c267c679d329db7c1a368c6f9